### PR TITLE
Camelize issues

### PR DESCRIPTION
**Pull request by hockeybuggy**
This is related to #15 

This PR:

- Creates test for the camelize method
- Strips leading and trailing space or `camelize`ed strings

### CODE REVIEW

File: pyactiveresource/util.py [hunk 1]
```diff
@@ -196,7 +196,9 @@ def camelize(word):
         The modified string.
     """
     return ''.join(w[0].upper() + w[1:]
-                   for w in re.sub('[^A-Z^a-z^0-9^:]+', ' ', word).split(' '))
+        for w
+        in re.sub('[^A-Z^a-z^0-9^:]+', ' ', word.strip()
+        ).split(' '))
```
Comment (kevinhughes27) [inline pyactiveresource/util.py]: I am not sure I agree with the multi line use here, is it really any easier to read?

**Comment by kevinhughes27** [comment]
fix looks good, thanks for the PR

**Comment by kevinhughes27** [comment]
@dylanahsmith can you have a look here too

File: pyactiveresource/util.py (hunk 1 above)
Comment (hockeybuggy) [inline pyactiveresource/util.py]: Hahaha. Nope. It's really not much easier to read but it was 107 cols wide when not split onto many lines. It felt like a bit much.

    return ''.join(w[0].upper() + w[1:] for w in re.sub('[^A-Z^a-z^0-9^:]+', ' ', word.strip()).split(' '))¬

I don't mind making it all one line if you would prefer? @kevinhughes27

Comment (kevinhughes27) [inline pyactiveresource/util.py]: its kind of lose lose lol. I'd vote one line by a small margin

Comment (hockeybuggy) [inline pyactiveresource/util.py]: As you wish. :stars:

File: pyactiveresource/util.py [hunk 2]
```diff
@@ -196,7 +196,7 @@ def camelize(word):
         The modified string.
     """
     return ''.join(w[0].upper() + w[1:]
-                   for w in re.sub('[^A-Z^a-z^0-9^:]+', ' ', word).split(' '))
+                   for w in re.sub('[^A-Z^a-z^0-9^:]+', ' ', word.strip()).split(' '))
```
Comment (dylanahsmith) [inline pyactiveresource/util.py]: You should strip the result of re.sub instead of the input to it, since it could also add leading spaces.

Comment (hockeybuggy) [inline pyactiveresource/util.py]: Great catch. Improved the tests to handle that case. Thanks.

**Comment by brandonkane** [comment]
What is the process for cutting a new release of pyactiveresource and the shopify python api now that this has been fixed?

**Comment by dylanahsmith** [comment]
There isn't really a process.  We should make a release.

**Comment by kevinhughes27** [comment]
I just pushed release of pyactiveresource - do we need a new one for shopify_python_api as well?

**Comment by brandonkane** [comment]
yes please - this bug was originally found when trying to access a store's data via the shopify_python_api and traced down to pyactiveresource

**Comment by kevinhughes27** [comment]
done! 2.1.5 is your new version with the bump!

**Comment by brandonkane** [comment]
Sweet - thanks everyone!

**Pull request merged**